File: ios/GitaVani/GitaVani/Theme/AppTheme.swift
import SwiftUI

struct AppTheme {
    let name: String
    let displayName: String
    let backgroundColor: Color
    let primaryTextColor: Color
    let secondaryTextColor: Color
    let accentColor: Color
    let cardBackgroundColor: Color
}

extension AppTheme {
    static let sattva = AppTheme(
        name: "sattva",
        displayName: "Sattva",
        backgroundColor: Color(red: 1.0, green: 1.0, blue: 1.0),
        primaryTextColor: Color(red: 0.102, green: 0.102, blue: 0.102),
        secondaryTextColor: Color(red: 0.4, green: 0.4, blue: 0.4),
        accentColor: Color(red: 0.0, green: 0.502, blue: 0.502),
        cardBackgroundColor: Color(red: 0.96, green: 0.96, blue: 0.96)
    )

    static let parchment = AppTheme(
        name: "parchment",
        displayName: "Parchment",
        backgroundColor: Color(red: 0.961, green: 0.941, blue: 0.91),
        primaryTextColor: Color(red: 0.243, green: 0.173, blue: 0.11),
        secondaryTextColor: Color(red: 0.478, green: 0.396, blue: 0.322),
        accentColor: Color(red: 0.757, green: 0.471, blue: 0.09),
        cardBackgroundColor: Color(red: 0.929, green: 0.906, blue: 0.867)
    )

    static let dusk = AppTheme(
        name: "dusk",
        displayName: "Dusk",
        backgroundColor: Color(red: 0.11, green: 0.11, blue: 0.18),
        primaryTextColor: Color(red: 0.91, green: 0.894, blue: 0.863),
        secondaryTextColor: Color(red: 0.659, green: 0.596, blue: 0.502),
        accentColor: Color(red: 0.831, green: 0.659, blue: 0.263),
        cardBackgroundColor: Color(red: 0.157, green: 0.157, blue: 0.231)
    )

    static let lotus = AppTheme(
        name: "lotus",
        displayName: "Lotus",
        backgroundColor: Color(red: 1.0, green: 0.961, blue: 0.902),
        primaryTextColor: Color(red: 0.29, green: 0.11, blue: 0.11),
        secondaryTextColor: Color(red: 0.47, green: 0.37, blue: 0.27),
        accentColor: Color(red: 0.86, green: 0.33, blue: 0.0),
        cardBackgroundColor: Color(red: 0.976, green: 0.929, blue: 0.863)
    )

    static let all: [AppTheme] = [.sattva, .parchment, .dusk, .lotus]

    static func named(_ name: String) -> AppTheme {
        all.first { $0.name == name } ?? .sattva
    }
}

File: android/GitaVani/app/src/main/java/com/nikhilsi/gitavani/theme/AppTheme.kt
package com.nikhilsi.gitavani.theme

import androidx.compose.ui.graphics.Color

data class AppTheme(
    val name: String,
    val displayName: String,
    val backgroundColor: Color,
    val primaryTextColor: Color,
    val secondaryTextColor: Color,
    val accentColor: Color,
    val cardBackgroundColor: Color
) {
    val isDark: Boolean get() = name == "dusk"

    companion object {
        val Sattva = AppTheme(
            name = "sattva",
            displayName = "Sattva",
            backgroundColor = Color(1.0f, 1.0f, 1.0f),
            primaryTextColor = Color(0.102f, 0.102f, 0.102f),
            secondaryTextColor = Color(0.4f, 0.4f, 0.4f),
            accentColor = Color(0.0f, 0.502f, 0.502f),
            cardBackgroundColor = Color(0.96f, 0.96f, 0.96f)
        )

        val Parchment = AppTheme(
            name = "parchment",
            displayName = "Parchment",
            backgroundColor = Color(0.961f, 0.941f, 0.91f),
            primaryTextColor = Color(0.243f, 0.173f, 0.11f),
            secondaryTextColor = Color(0.478f, 0.396f, 0.322f),
            accentColor = Color(0.757f, 0.471f, 0.09f),
            cardBackgroundColor = Color(0.929f, 0.906f, 0.867f)
        )

        val Dusk = AppTheme(
            name = "dusk",
            displayName = "Dusk",
            backgroundColor = Color(0.11f, 0.11f, 0.18f),
            primaryTextColor = Color(0.91f, 0.894f, 0.863f),
            secondaryTextColor = Color(0.659f, 0.596f, 0.502f),
            accentColor = Color(0.831f, 0.659f, 0.263f),
            cardBackgroundColor = Color(0.157f, 0.157f, 0.231f)
        )

        val Lotus = AppTheme(
            name = "lotus",
            displayName = "Lotus",
            backgroundColor = Color(1.0f, 0.961f, 0.902f),
            primaryTextColor = Color(0.29f, 0.11f, 0.11f),
            secondaryTextColor = Color(0.47f, 0.37f, 0.27f),
            accentColor = Color(0.86f, 0.33f, 0.0f),
            cardBackgroundColor = Color(0.976f, 0.929f, 0.863f)
        )

        val all = listOf(Sattva, Parchment, Dusk, Lotus)

        fun named(name: String): AppTheme =
            all.firstOrNull { it.name == name } ?: Sattva
    }
}
